sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExtendedExec.scala

```diff
@@ -0,0 +1,201 @@
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.jdk.CollectionConverters._
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.analysis.ResolvedPartitionSpec
+import org.apache.spark.sql.catalyst.catalog.CatalogTableType
+import org.apache.spark.sql.catalyst.catalog.ExternalCatalogUtils.escapePathName
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Cast, Literal}
+import org.apache.spark.sql.catalyst.util.{quoteIdentifier, StringUtils}
+import org.apache.spark.sql.connector.catalog.{CatalogV2Util, Identifier, SupportsPartitionManagement, Table, TableCatalog}
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits._
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.sql.execution.datasources.v2.DataSourceV2Implicits.TableHelper
+import org.apache.spark.sql.types.{StringType, StructType}
+
+/**
+ * Physical plan node for showing tables without partition, Show the information of tables.
+ */
+case class ShowTablesExtendedExec(
+    output: Seq[Attribute],
+    catalog: TableCatalog,
+    namespace: Seq[String],
+    pattern: String) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val rows = new ArrayBuffer[InternalRow]()
+
+    // fetch tables
+    // TODO We need a new listTable overload that takes a pattern string.
+    val tables = catalog.listTables(namespace.toArray)
+    tables.map { tableIdent =>
+      if (Option(pattern).forall(StringUtils.filterPattern(
+        Seq(tableIdent.name()), _).nonEmpty)) {
+        val table = catalog.loadTable(tableIdent)
+        val information = getTableDetails(catalog.name, tableIdent, table)
+        rows += toCatalystRow(tableIdent.namespace().quoted, tableIdent.name(), false,
+          s"$information\n")
+        }
+      }
+
+    // fetch views, includes: view, global temp view, local temp view
+    val sessionCatalog = session.sessionState.catalog
+    val (namespaceExists, db) = namespace match {
+      case Seq(db) =>
+        (sessionCatalog.databaseExists(db), Some(db))
+      case _ =>
+        (false, None)
+    }
+    if (namespaceExists) {
+      val views = sessionCatalog.listViews(db.get, pattern)
+      views.map { viewIdent =>
+        val tableName = viewIdent.table
+        val isTemp = sessionCatalog.isTempView(viewIdent)
+        val view = sessionCatalog.getTempViewOrPermanentTableMetadata(viewIdent)
+        val information = view.simpleString
+        rows += toCatalystRow(db.get, tableName, isTemp, s"$information\n")
+      }
+    }
+
+    rows.toSeq
+  }
+
+  private def getTableDetails(
+      catalogName: String,
+      identifier: Identifier,
+      table: Table): String = {
+    val results = new mutable.LinkedHashMap[String, String]()
+
+    results.put("Catalog", catalogName)
+
+    if (!identifier.namespace().isEmpty) {
```

Comment:
I think we should always put a namespace field, empty string if it's Nil

Reply:
Okay